prac_09/unreliable_car.py

```diff
@@ -0,0 +1,22 @@
+"""
+CP1404/CP5632 Practical - unreliable car.
+"""
+from prac_09.car import Car
+from random import uniform
+
+
+class UnreliableCar(Car):
+    """Represent class unreliable car"""
+
+    def __init__(self, name, fuel, reliability=0.0):
+        """Initialise a Taxi instance, based on parent class Car."""
+        super().__init__(name, fuel)
+        self.reliability = reliability
+
+    def drive(self, distance):
+        """Drive car if random number is greater than reliability."""
+        random_number = uniform(0, 100)
+        if random_number >= self.reliability:
```

Comment:
You seem to have the percentage mixed around reliability is a percentage such as 15% which mean 15% of the time the car works which can be represented by number (1-15 out of 100 == 15% out of 100) so to have the car break down weed we to compare the number 1-100 and if any is greater than 15 (equal was my choice) it breaks down this represents the 85% chance or (16-100 == 85% out of 100)

In you case of random number was less than reliability this would be the reverse, cars with a 15% reliability would run 85% of the time.

Though this also depends on what comes after the if statement as:
if number < reliabililty 
 then drive
 
 and 
 if number >= reliability
 then dont drive 
 
 they are the equivilent statements.

Reply:
You seem to have the percentage mixed around reliability is a percentage such as 15% which mean 15% of the time the car works which can be represented by number (1-15 out of 100 == 15% out of 100) so to have the car break down weed we to compare the number 1-100 and if any is greater than 15 (equal was my choice) it breaks down this represents the 85% chance or (16-100 == 85% out of 100)

In you case of random number was less than reliability this would be the reverse, cars with a 15% reliability would run 85% of the time.

Though this also depends on what comes after the if statement as:
if number < reliabililty 
 then drive
 
 and 
 if number >= reliability
 then dont drive 
 
 they are the equivilent statements.